### DISCUSSION TITLE
gltfNode structure as output & for loop over that output

### DISCUSSION THREAD

**Discussion by noahabood** [cables-gl/cables_docs#174] [Feature Requests] [answered]
I need to extract all the nodes from glft file, it is exhausting to extract every separate node from structure when you have 1000 geometry in glft file, and it would be great if there block for iterating over the output from structure
it would be nice if there is for loop:

for x in structure:
      return x
      
![image](https://user-images.githubusercontent.com/100239828/155222266-ee594279-1058-4f20-87b7-794ebbcd0f2b.png)

**Comment by pandrr** [comment, marked as answer]
gltfscene has a json output port which contains an array of all nodes (e.g. name of the node) which can be iterated over.
you then could extracted the name to use it with gltfnode to draw it...

what do you want to do with the nodes/geometries ?

**Reply by noahabood** [reply]
Hi pander,
vielen Dank,
it seems like a good idea, but how can I extract only the names of the arrays in cabels gl to feed them into gltfnode. I am new to cabels gl I tried some nodes in array, but it did not work.
I am trying to attach some information to geometry from other json file, something like in screenshot, i am tring to connect geometry of a building to its information,to have something like "BIM viewer" in web gl 
![image](https://user-images.githubusercontent.com/100239828/156460608-cff9aee4-190f-4c4c-b3ca-070a92cf2d47.png)

thanks for your help
